django-backend/fecfiler/scha_transactions/models.py

```diff
@@ -69,3 +79,39 @@ class SchATransaction(SoftDeleteModel, CommitteeOwnedModel):
 
     class Meta:
         db_table = "scha_transactions"
+        indexes = [
+            models.Index(fields=['transaction_id'])
+        ]
+
+    # This is intended to be useable without instantiating a transaction object
+    @staticmethod
+    def check_for_uid_conflicts(uid): # noqa
+        return len(SchATransaction.objects.filter(transaction_id=uid)) > 0
+
+    def generate_unique_transaction_id(self):
+        u = uuid.uuid4()
+        hex_id = u.hex.upper()
+        # Take 20 characters from the end, skipping over the 20th char from the right,
+        # which is the version number (uuid4 -> "4")
+        unique_id = hex_id[-21] + hex_id[-19:]
+
+        attempts = 0
+        while SchATransaction.check_for_uid_conflicts(unique_id):
+            unique_id = str(u.hex).upper()[-20:]
```

Comment:
I think `u.hex` returns the same value every time you call it.  We might need to make a new `uuid.uuid4()` to get a new id

Reply:
Good catch.  I'm disappointed that I caught that in the migration and didn't also catch it in the model.  Took the opportunity to reduce the code duplication as well by wrapping the whole unique_id generation in its own method.